### PR TITLE
workflows: update and pin node versions

### PR DESCRIPTION
**Pull request by gurgunday**

### CODE REVIEW

**Comment by mcollina** [comment]
windows on node 22 must be skipped for now

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -17,7 +17,8 @@ on:
 
 jobs:
   test:
-    uses: fastify/workflows/.github/workflows/plugins-ci.yml@v4.1.0
+    uses: fastify/workflows/.github/workflows/plugins-ci.yml@v4.2.1
```
Comment (climba03003) [inline .github/workflows/ci.yml:20]: I believe we can release `v4`, because the `node` version is now configurable.

Comment (gurgunday) [reply .github/workflows/ci.yml:20]: We’d have to configure every plugin manually, so not sure

I’m pinning libraries right now as they are fastify independent

Comment (gurgunday) [reply .github/workflows/ci.yml:20]: Yes it does, but the default tests for 18, 20, 21

main branches still test for 14, 16, so they would all need manual modification, no?

Comment (gurgunday) [reply .github/workflows/ci.yml:20]: But yeah not saying we cannot do it

Comment (climba03003) [reply .github/workflows/ci.yml:20]: Eventually, we need to update all of them back to `v4` tag.
We can do this earlier if we release.

Comment (gurgunday) [reply .github/workflows/ci.yml:20]: Yeah actually could be good as even next versions currently have hardcoded versions... which in not ideal

**Pull request closed without merging**